packages/headless/src/state/search-app-state.ts

```diff
@@ -41,6 +42,7 @@ export type SearchParametersState = FacetSection &
   AdvancedSearchQueriesSection &
   PaginationSection &
   SortSection &
+  ProductListingSortSection &
```

Comment:
:thinking: Not sure it makes sense to have a "ProductListingSortSection" in the "Search App State". 

On the other hand, we need this, otherwise it won't be picked up by the History :thinking: 

@samisayegh any opinion on this?

Reply:
The `ProductListingSortSection` should not be a part of `SearchAppState` imo since the search use-case does not leverage it. As a dev, it would not be relevant to me to see `ProductListing...` (or any other use-case names) in intellisense when building a search page.

I would leave out history support for now. We will probably need to refactor to make it possible to extend History, while still keep accurate typing. Another option could be to loosen the typing to `any`. Which ever path we take, I think this is best left for a separate PR.

Reply:
Alright, I removed it all in the last commit :)